libgalois/src/PropertyGraph.cpp

```diff
@@ -551,3 +551,124 @@ katana::SortNodesByDegree(katana::PropertyGraph* pg) {
 
   return katana::ResultSuccess();
 }
+
+katana::Result<std::unique_ptr<katana::PropertyGraph>>
+katana::CreateSymmetricGraph(katana::PropertyGraph* pg) {
+  const GraphTopology& topology = pg->topology();
+  auto out_dests_view_result =
+      ConstructPropertyView<UInt32Property>(topology.out_dests.get());
+  if (!out_dests_view_result) {
+    return out_dests_view_result.error();
+  }
+  //Original property graph destination vector
+  auto out_dests_view = std::move(out_dests_view_result.value());
+
+  // New symmetric graph topology
+  katana::LargeArray<uint64_t>* out_indices_symmetric =
+      new katana::LargeArray<uint64_t>();
+  katana::LargeArray<uint32_t>* out_dests_symmetric =
+      new katana::LargeArray<uint32_t>();
+  katana::LargeArray<uint64_t> out_indices_symmetric_temp;
+
+  out_indices_symmetric->allocateInterleaved(topology.num_nodes());
+  out_indices_symmetric_temp.allocateInterleaved(topology.num_nodes());
+
+  auto edge_range = topology.edge_range(0);
```

Comment:
- `edge_range` is deprecated; use `edges` instead, e.g., `edges(node).begin()`
- what if the graph has no nodes?